src/main/java/core/basesyntax/impl/ProcessDataImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.impl;
+
+import core.basesyntax.service.ProcessData;
+import core.basesyntax.transactor.FruitTransaction;
+import core.basesyntax.transactor.Operation;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class ProcessDataImpl implements ProcessData {
+    private static final String COMMA = ",";
+    private static final int HEAD_INFO_INDEX = 1;
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_TYPE_INDEX = 1;
+    private static final int INDEX_OF_AMOUNT = 2;
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
+    @Override
+    public List<FruitTransaction> process(String inputData) {
+        String[] records = inputData.split(LINE_SEPARATOR);
+        return Stream.of(records).skip(HEAD_INFO_INDEX)
+                .map(this::convertDataToTransaction).toList();
+    }
+
+    private Operation getOperation(String code) {
+        return Stream.of(Operation.values())
+                .filter(o -> o.getCode().equals(code))
+                .findFirst().orElseThrow(() -> new RuntimeException(
+                        "No operation following code: " + code));
+    }
+
+    private FruitTransaction convertDataToTransaction(String inputDataLine) {
+        String[] data = inputDataLine.split(COMMA);
+        return new FruitTransaction(getOperation(data[OPERATION_INDEX].trim()),
+                data[FRUIT_TYPE_INDEX].trim(), Integer.parseInt(data[INDEX_OF_AMOUNT].trim()));
```

Comment:
remove trim. Better validate  that amount number and throw an exception for invalid input data